src/sciline/pipeline.py

```diff
@@ -72,6 +72,20 @@ def _is_compatible_type_tuple(
     return True
 
 
+def _qualname(obj: Any) -> Any:
+    return (
+        obj.__qualname__ if hasattr(obj, '__qualname__') else obj.__class__.__qualname__
+    )
+
+
+def _kind_of_provider(p: Callable[..., Any]) -> str:
+    if _qualname(p) == f'{_qualname(Pipeline.__setitem__)}.<locals>.<lambda>':
+        return 'parameter'
+    if _qualname(p) == f'{_qualname(Pipeline.set_param_table)}.<locals>.<lambda>':
```

Comment:
Will it be better to make a dummy ``Pipeline`` with a parameter and a table
and retrieve their names, and use them to check these instead?

Or if you add the passing case unit test I mentioned, it might be fine like this.

Reply:
I don't quite understand what you mean. Do you mean to compute the expected `_qualname`s instead of hardcoding something like `f'{_qualname(Pipeline.set_param_table)}.<locals>.<lambda>'`?

Reply:
No sth like.. 
```python 
_DUMMY_PIPELINE = sl.Pipeline(params={int: 0})
_PARAMETER_PROVIDER_NAME = _DUMMY_PIPELINE._get_provider(int)[0].__qualname__
```